src/tests.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include "prefelem.h"
+
+#include <cstdio>
+#include <exception>
+
+#define ASSERT(COND)                                                                            \
+    do {                                                                                        \
+        if (!(COND)) {                                                                          \
+            ::std::fprintf(stderr, "Assertion failed: %s in %s:%d", #COND, __FILE__, __LINE__); \
+            ::std::terminate();                                                                 \
+        }                                                                                       \
+    } while (0)
+
+void TestInMemServices();
+void TestPreferredItemService();
+
+namespace prefelem
+{
+inline bool operator==(const Item& x, const Item& y)
+{
+    bool result = x.id == y.id;
+    ASSERT(result == (x.bricksAndCounts == y.bricksAndCounts));
```

Comment:
I just noticed this error but I didn't want to fix it since coding time is over.
It's an error because different items might have the same bricks.
```suggestion
    ASSERT(!result || (x.bricksAndCounts == y.bricksAndCounts));
```